### ISSUE TITLE
Add support for Lombok's @EqualsAndHashCode(cacheStrategy = LAZY)

### ISSUE THREAD

**Issue by janeisklar** [jqno/equalsverifier#460]
**Is your feature request related to a problem? Please describe.**
The following class currently can't be tested using equalsverifier:

```java
@RequiredArgsConstructor
@EqualsAndHashCode(cacheStrategy = EqualsAndHashCode.CacheStrategy.LAZY)
public class CachedHashCode {
  private final String foo;
}
```

Equalsverifier fails with the following error message: `hashCode relies on $hashCodeCache, but equals does not.`. 

**Describe the solution you'd like**
I want to be able to test the hashcode method that is created by Lombok when using its lazy caching strategy.

**Describe alternatives you've considered**
Using `.withCachedHashCode` would seem like an obvious option to work around this, but this unfortunately only works with private methods and thus fails with `Cached hashCode: Could not find calculateHashCodeMethod: must be 'private int hashCode()'`.

**Additional context**
I will provide a pull request with an proposal on how to solve this in a bit.

**Comment by jqno** [comment]
Thank you for the nice PR, and for even updating the docs! I'll let you know when I release this.

**Comment by jqno** [comment]
I've just released version 3.7!

**Comment by janeisklar** [comment]
Thanks again for the extremely fast response time and a great utility:)
